errors.go

```diff
@@ -7,6 +7,10 @@ import (
 	"github.com/docker/distribution/digest"
 )
 
+// ErrManifestNotModified is returned when a conditional manifest GetByTag
+// returns nil due to the client indicating it has the latest version
+var ErrManifestNotModified = fmt.Errorf("manifest not modified")
```

Comment:
This could use `errors.New` since there's no string formatting involved. Same with `ErrUnsupported` below.